cdc/kv/shared_stream.go

```diff
@@ -169,7 +169,8 @@ func (s *requestedStream) run(ctx context.Context, c *SharedClient, rs *requeste
 		}
 	}()
 
-	cc, err := c.grpcPool.Connect(ctx, rs.storeAddr)
+	g, gctx := errgroup.WithContext(ctx)
+	cc, err := c.grpcPool.Connect(gctx, rs.storeAddr)
```

Comment:
Please add some comments to explain that gctx is used to control grpc clientStream, and that the underlying grpc conn that has been successfully established is unaffected when gctx is cancelled.